### PR TITLE
Initial version of MS acquisition geometry storage

### PR DESCRIPTION
**Pull request by iprotsyuk**
The PR enables sm-engine to store a json object in the DB, which describes the geometry of mass spec acquisition in terms of pixel layout and pixel sizes.

Some ideas are maybe not implemented in the best way. I'll highlight them in the diff shortly.

The acquisition geometry can be stored, but nothing is implemented to retrieve it, e.g. for the web app. I'm planning to add necessary endpoints later, likely within this PR, after the current code is reviewed. I also haven't yet updated any tests that were affected by the changes.

### CODE REVIEW

File: conf/config.json.template [hunk 1]
```diff
@@ -39,10 +39,16 @@
   },
   "aws": {
     "aws_access_key_id": "{{ aws_access_key_id }}",
-    "aws_secret_access_key": "{{ aws_secret_access_key }}"
+    "aws_secret_access_key": "{{ aws_secret_access_key }}",
+    "geo_preference": "eu-west-1"
```
Comment (intsco) [inline conf/config.json.template]: @iprotsyuk So far, aws region is specified in Ansible for all components of SM Platform

File: sm/engine/acq_geometry_factory.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+def geometry_section(section_name):
+    """ Class decorator adding static property @section_name with the input value to a decorated class
+    """
+    def modifier(cls):
+        cls.section_name = section_name
+        return cls
+    return modifier
+
+
+class ACQ_GEOMETRY_KEYS(object):
```
Comment (intsco) [inline sm/engine/acq_geometry_factory.py]: @iprotsyuk just as a comment, names tuples can also be used for grouping constants
https://stackoverflow.com/questions/29792189/grouping-constants-in-python#29794517

File: sm/engine/dataset.py [hunk 3]
```diff
@@ -88,6 +97,13 @@ def save(self, db, es, queue=None):
         if queue:
             queue.publish({'ds_id': self.id, 'status': self.status}, SM_DS_STATUS)
 
+    def save_acq_geometry_from_file(self, db, ms_file_path):
```
Comment (intsco) [inline sm/engine/dataset.py]: @iprotsyuk can we rename it to "import_acq_geometry_from_file"

Comment (iprotsyuk) [reply sm/engine/dataset.py]: yep, sounds better

File: sm/engine/acq_geometry_factory.py (hunk 2 above)
Comment (iprotsyuk) [reply sm/engine/acq_geometry_factory.py]: yep, I'm aware of these. The use of `class` looks better to me in this particular case. I'd keep it this way, unless there're any other objections

File: conf/config.json.template (hunk 1 above)
Comment (intsco) [reply conf/config.json.template]: ok, we do need this new parameter
for consistency please rename it to `aws_region`

**Comment by intsco** [comment]
@iprotsyuk Please make sure the code passes existing tests. It would be nice to have tests for the new classes as well. Once it's done, we'll do merge

**Comment by intsco** [comment]
@iprotsyuk By the way, I didn't see a `parser_factory` for LC data

**Comment by iprotsyuk** [comment]
@intsco, yes, I'll add new tests.
As for the LC data parser factory, would you suggest to keep it in the master branch or in a separate "LC branch" instead?

**Comment by intsco** [comment]
@iprotsyuk all the code should eventually be merged into master. Otherwise, it will be quite difficult to handle cross updates of both branches.

**Comment by iprotsyuk** [comment]
@intsco, alright. I wasn't sure about this, so didn't add the LC implementation. Will add it to this PR as well

**Comment by iprotsyuk** [comment]
@intsco, I think all the discussed points are addressed in my latest commits

**Comment by intsco** [comment]
@iprotsyuk the next would be pulling all the changes to master and testing it components together first locally. Have you finished with changes of the graphql and web app?

**Comment by iprotsyuk** [comment]
@intsco, not yet. I thought these components could be merged independently of each other provided no existing tests fail and new tests added where appropriate.
But if for you it's easier to merge everything at once, we can wait till other layers are ready.

**Pull request merged**